src/base/misc/dos2linux.c

```diff
@@ -753,7 +758,10 @@ static int check_write_pagefault(dosaddr_t addr, void *uaddr, uint32_t op, int l
     handler(addr, 6 + dpmi_read_access(addr), op, len);
     return 1;
   }
-  if (!e_querymprot(addr) && !memcheck_is_rom(addr))
+  if (e_querymprot(addr))
+    /* handle invalidations for writes to JIT protected pages */
+    handler(addr, -1, op, len);
```

Comment:
Or for example you can call
e_invalidate() right here?

Reply:
At the moment, yes, but the idea here is that codegen-sim.c hooks this handler, and does this instead:
```
       if (err == -1) {
               /* err = -1 is special, hitting JIT protected page */
               if (e_querymark(addr, len))
                       InvalidateNodeRange(addr, len, currentIG);
               return;
       }
```
where currentIG is set like in the present #2519 draft but as a static variable.